### PR TITLE
Build: Get rid of LD_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by NiLuJe**
Companion PR to https://github.com/koreader/koreader-base/pull/1638

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10705)

### CODE REVIEW

File: platform/pocketbook/koreader.app [hunk 1]
```diff
@@ -4,6 +4,10 @@ export LC_ALL="en_US.UTF-8"
 # working directory of koreader
 KOREADER_DIR="/mnt/ext1/applications/koreader"
 
+# load our own shared libraries if possible, solely because we don't control InkView, and we'd rather not it have load duplicate system libs...
+# (We handle this via DT_RPATH for our own stuff).
+export LD_LIBRARY_PATH="${KOREADER_DIR}/libs:${LD_LIBRARY_PATH}"
```
Comment (benoit-pierre) [inline platform/pocketbook/koreader.app:9]: Unless the InkView library has a RPATH/RUNPATH entry, our executable RPATH entry should still get priority, no?

Comment (NiLuJe) [reply platform/pocketbook/koreader.app:9]: I frankly don't trust it not to do something funky, so I'd rather be vewy, vewy cautious ;).

Comment (NiLuJe) [reply platform/pocketbook/koreader.app:9]: (Plus, I don't know if the base system doesn't have a funky system-wide LD_LIBRARY_PATH, or, worse, an LD_PRELOAD that could do nasty things)

Comment (benoit-pierre) [reply platform/pocketbook/koreader.app:9]: And combined with removing the `env -u LD_LIBRARY_PATH ` prefix in `runCommand`?

Comment (benoit-pierre) [reply platform/pocketbook/koreader.app:9]: > (Plus, I don't know if the base system doesn't have a funky system-wide LD_LIBRARY_PATH, or, worse, an LD_PRELOAD that could do nasty things)

You could unset those, I think `LD_PRELOAD` still get priority anyway, no?

Comment (NiLuJe) [reply platform/pocketbook/koreader.app:9]: > And combined with removing the `env -u LD_LIBRARY_PATH ` prefix in `runCommand`?

That's done in this PR? Not sure what you were thinking of there ;).


> You could unset those, I think `LD_PRELOAD` still get priority anyway, no?

Yeah, we're can't do anything about a preload hack.

Unsetting LD_LIBRARY_PATH instead could possibly achieve the same thing, but, again, this is a weird platform that I can't test nor trust, so I'm really going brute-force on purpose ;).

Comment (benoit-pierre) [reply platform/pocketbook/koreader.app:9]: > >    And combined with removing the env -u LD_LIBRARY_PATH prefix in runCommand?
>
> That's done in this PR? Not sure what you were thinking of there ;).

This change in `frontend/device/sdl/device.lua`:
```diff
 local function runCommand(command)
-     local env = jit.os ~= "OSX" and 'env -u LD_LIBRARY_PATH ' or ""
-     return os.execute(env..command) == 0
+     return os.execute(command) == 0
 end
```

Comment (benoit-pierre) [reply platform/pocketbook/koreader.app:9]: Our library directory now takes priority when calling external executable on pocketbook.

Comment (benoit-pierre) [inline platform/pocketbook/koreader.app:9]:
```suggestion
export LD_LIBRARY_PATH="${KOREADER_DIR}/libs${LD_LIBRARY_PATH+:}${LD_LIBRARY_PATH}"
```

Comment (NiLuJe) [reply platform/pocketbook/koreader.app:9]: That's always been the case, because that snippet is an SDL-only codepath, so it never applied to PB.

Comment (NiLuJe) [reply platform/pocketbook/koreader.app:9]: Is that documented somewhere?

That idiom is *very* common as far as I'm aware, and I wasn't aware that a set but empty LD_LIBRARY_PATH meant $PWD (or is that just an empty string after a colon?).

Comment (benoit-pierre) [reply platform/pocketbook/koreader.app:9]: Fully empty, no, but with `LD_LIBRARY_PATH=foobar:` the current directory is searched too.

Comment (NiLuJe) [reply platform/pocketbook/koreader.app:9]: Well that's... sneaky and weird.

Let's hope early busybox versions were actually POSIX compliant and won't balk at the `+` parameter expansion ;).

**Comment by NiLuJe** [comment]
(Rebase me)

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 15 of 16 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/10705)* status: all files reviewed, 2 unresolved discussions (waiting on @benoit-pierre and @Frenzie)

**Pull request merged**